### ISSUE TITLE
vmctl can't read downsampled Thanos data blocks

### ISSUE THREAD

**Issue by vishnubraj** [VictoriaMetrics/VictoriaMetrics#1101]
VMCTL not importing Thanos data anymore. Getting below-mentioned error when we try to import.

./vmctl-prod prometheus --prom-snapshot 01EZ7XXXXXXXXXX7H2S/chunks --vm-addr http://vminsert2a.sjc2:8480
Prometheus import mode
Prometheus snapshot stats:
  blocks found: 0;
  blocks skipped by time filter: 0;
  min time: 0 (1970-01-01T00:00:00Z);
  max time: 0 (1970-01-01T00:00:00Z);
  samples: 0;
  series: 0.
2021/03/01 15:00:10 found no blocks to import

**Comment by hagen1778** [comment]
@vishnubraj is it possible to get the obfuscated/non-sensitive example of data block you're trying to import? Or any procedure which I can follow to reproduce it?
Would also appreciate details of Thanos installation - the version, components and configs. Thank you!

**Comment by vishnubraj** [comment]
Sorry, @hagen1778  I won't be able to provide the data block. 

I have the last 1 year of data in S3. It's compacted as 2 weeks block. I copied a block with raw data from s3 to my local server and tried importing to Victoria metrics using vmctl. 

**Thanos compact**
version  = 0.17.2
config  
`/usr/bin/thanos compact --data-dir /data --objstore.config-file=/etc/prometheus/s3_ob_config_file.yaml --retention.resolution-raw=2y --retention.resolution-5m=5y --retention.resolution-1h=10y`

Thanos Sidecar
version =  0.15.0-rc.0
Config
`/usr/bin/thanos sidecar --tsdb.path /instance1 --http-address=0.0.0.0:19090 --prometheus.url http://localhost:9090 --grpc-address=0.0.0.0:10912 --objstore.config-file /etc/prometheus/s3_aws_access.yml`

Thanos store
version =  0.15.0-rc.0
Config
`/usr/bin/thanos store --data-dir=/data/store --max-time=-90d --min-time=2019-07-29T00:00:00Z --objstore.config-file /etc/prometheus/s3_aws_access.yml --log.level=debug --store.grpc.series-max-concurrency=100 --http-address=0.0.0.0:10902 --grpc-address=0.0.0.0:10914`

Thanos query
version =  0.15.0-rc.0
Config
`/usr/bin/thanos query --http-address=0.0.0.0:19192 --store=prometheus1a.sjc2:10912 --store=127.0.0.1:10914 --query.replica-label=replica --query.timeout=2m --query.max-concurrent=12 --grpc-address=0.0.0.0:10913`

**Comment by hagen1778** [comment]
@vishnubraj have you tried https://github.com/timescale/promscale/tree/master/cmd/prom-migrator as a workaround? I'm not familiar with this tool yet, but docs promise to support such a migration.

**Comment by vishnubraj** [comment]
Yes, @hagen1778  I tried this, but Thanos store times out when the migration starts. 
It tries to query the first one-minute data and Thanos store times out with the below error. I am going to raise a ticket with them.

`level=debug ts=2021-03-02T14:32:54.694630715Z caller=bucket.go:993 msg="stats query processed" stats="&{blocksQueried:1 postingsTouched:1 postingsTouchedSizeSum:8348668 postingsToFetch:0 postingsFetched:1 postingsFetchedSizeSum:8348668 postingsFetchCount:1 postingsFetchDurationSum:173430545 cachedPostingsCompressions:1 cachedPostingsCompressionErrors:0 cachedPostingsOriginalSizeSum:8348668 cachedPostingsCompressedSizeSum:948850 cachedPostingsCompressionTimeSum:39563631 cachedPostingsDecompressions:0 cachedPostingsDecompressionErrors:0 cachedPostingsDecompressionTimeSum:0 seriesTouched:2087166 seriesTouchedSizeSum:7759035328 seriesFetched:2087166 seriesFetchedSizeSum:7787482272 seriesFetchCount:1 seriesFetchDurationSum:77909345526 chunksTouched:1995842 chunksTouchedSizeSum:291363676 chunksFetched:1995842 chunksFetchedSizeSum:145903464063 chunksFetchCount:55948 chunksFetchDurationSum:866512903290886 getAllDuration:243967687800 mergedSeriesCount:979396 mergedChunksCount:983014 mergeDuration:0}" err="rpc error: code = Unknown desc = send series response: rpc error: code = Canceled desc = context canceled"`

**Comment by valyala** [comment]
FYI, [vmctl](https://docs.victoriametrics.com/vmctl.html) is able to migrate data from any system, which supports Prometheus remote_read protocol, to VictoriaMetrics, starting from [v1.85.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). See [these docs](https://docs.victoriametrics.com/vmctl.html#migrating-data-by-remote-read-protocol).

**Comment by f41gh7** [comment]
Closing as resolved
